content/en/tutorials/advanced/real-time-multiplayer.md

```diff
@@ -0,0 +1,440 @@
+
+~~~
+node server.js
+~~~
+
+You should see ‘Server started.’ Congratulations, you’re now running your very own server.
+
+![Server Started][2]
+
+## Setting up the Project
+
+Open up PlayCanvas and create a new project. First, you’ll want to create a new script called ‘socket.js’. Next, add it to your projects priority list.
+
+![Script Priority][3]
+
+Open the script and replace the contents of the file with [this.][4]
+
+Now we need to create a new script to handle the network logic. Create a new script called ‘Network.js’. We first need to create a connection to the server. We can do this by adding this line in the initialize method:
+
+~~~javascript~~~
+this.socket = io.connect(‘http://localhost:3000/');
+~~~
+
+‘http://localhost' is the address of your server and the ‘3000’ is the port. Since we’re connecting to our own computer, we use localhost. If you were hosting elsewhere this is what you'd want to change. Make sure the port you’re using to connect is the same as the one you set in the server file. The project is now setup to send and receive messages to/from the server.
+
+## Server and Client Communication
+
+The way you can send data between the client and server is with the socket connection we made earlier. To send data from the client, we use the emit function. Here’s an example:
+
+~~~javascript~~~
+this.socket.emit (‘playerJoined’, ‘John’);
+~~~
+
+This emits a message called ‘playerJoined’, with the data ‘John’. For the server to receive the message, we need to write in the server file:
+	
+~~~javascript~~~
+socket.on (‘playerJoined’, function (name) {
+	console.log (name);
+});
+~~~
+
+This will log whatever data is sent to the server when ‘playerJoined’ is emitted.
+
+For this demo, we’re aiming to have players move around with others in real time, so we'll need to create an environment. Start by create an entity to use as a ground, and add a collision box and static rigidbody.
+
+![Ground Entity][7]
+
+Next we’ll need a player to control. Create a new capsule and call it ‘Player’. add a dynamic rigidbody and collision box, and change the rigid body settings to match the picture below.
+
+![Player Entity][8]
+
+Duplicate the player entity and rename it as 'Other'. This is the entity we'll be using to simulate other players in the game.
+
+Add a script component to your player, and attach a new script called 'Movement.js':
+
+~~~javascript~~~
+pc.script.attribute('playerSpeed', 'number', 30, {displayName: 'Player Speed'});
+
+pc.script.create('Movement', function (app) {
+	// Creates a new Movement instance
+	var Movement = function (entity) {
+		this.entity = entity;
+		this.force = new pc.Vec3 ();
+	};
+
+	Movement.prototype = {
+		// Called every frame, dt is time in seconds since last update
+		update: function (dt) {
+			var forward = this.entity.forward;
+			var right = this.entity.right;
+            
+			x = 0;
+			z = 0;
```

Comment:
Put 'var' in front of x and y, otherwise they will be in global scope (probably not what you want).